govcd/adminvdc.go

```diff
@@ -52,15 +52,26 @@ var vdcVersionedFuncsV97 = vdcVersionedFuncs{
 	UpdateVdcAsync:   updateVdcAsyncV97,
 }
 
-// TODO: add a wrapper function to use newest available method when version is higher than currently handled
-// VDC function mapping by vDC version
+// vdcVersionedFuncsByVcdVersion is a map of VDC functions by vCD version
 var vdcVersionedFuncsByVcdVersion = map[string]vdcVersionedFuncs{
 	"vdc9.5":  vdcVersionedFuncsV95,
 	"vdc9.7":  vdcVersionedFuncsV97,
 	"vdc10.0": vdcVersionedFuncsV97,
 	"vdc10.1": vdcVersionedFuncsV97,
 }
 
+// getVdcVersionedFuncsByVdcVersion is a wrapper function that retrieves the requested versioned VDC function
+// When the wanted version does not exist in the map, it returns the latest available one.
+func getVdcVersionedFuncsByVdcVersion(version string) vdcVersionedFuncs {
+	highest := vdcVersionedFuncsV97
```

Comment:
Hm. Looks like we shouldn't be hard-coding this. What about choosing the last entry in the `vdcVersionedFuncsByVcdVersion` map instead? That should give you the highest version automatically.

Reply:
A map doesn't have a **last** entry.
We need to change the `highest` when we create a new VDC function.  I can add a comment about that in `vdcVersionedFuncsByVcdVersion`

Reply:
one possible variation is to add `highest/default` in the map and fetch it when specific fetch fails.

Reply:
Updated with default entry.

Reply:
> 	// If we add a new function to this list, we also need to update the "default" entry
	"default": vdcVersionedFuncsV97,

What about mentioning in the comment that this is the highest version for clarity?

Reply:
added